### DISCUSSION TITLE
Solar Generation Total

### DISCUSSION THREAD

**Discussion by LukeMc8** [nathanmarlor/foxess_modbus#88] [Q&A] [answered]
Just a quick question, the solar generation solar sensor. I'm guessing this is the combined generation of PV1 and PV2? If i wanted to add CT2 generation to this total which file would i need to edit? The stealthchesnut integration was a lot easier to read and i knew what to change, this one is working much better but i am struggling a little with editing it.

**Comment by canton7** [comment, marked as answer]
Yes, Solar Generation is reported directly by the inverter, and is PV1 + PV2.

Just as with the stealthchesnut integration, you can add your own template sensors. So for example you could create a new sensor which adds together PV1, PV2 and CT2, by editing your configuration.yaml and adding a `template:` section.

**Reply by LukeMc8** [reply]
Thanks for the help, I'm still learning but would adding something like the below to my configuration file add the 3 generations individually?

sensor: 
  - method: left
    name: pv1_sum
    unique_id: foxess_inv1_pv1_sum
    platform: integration
    round: 2
    source: sensor.fox_h1_pv1_power
    unit_time: h
  - method: left
    name: pv2_sum
    unique_id: foxess_inv1_pv2_sum
    platform: integration
    round: 2
    source: sensor.fox_h1_pv2_power
    unit_time: h
  - method: left
    name: ct2_sum
    unique_id: foxess_inv1_ct2_sum
    platform: integration
    round: 2
    source: sensor.fox_h1_ct2_meter
    unit_time: h

utility_meter:
  pv1_daily:
    unique_id: foxess_inv1_pv1_daily
    source: sensor.pv1_sum
    cycle: daily
  pv2_daily:
    unique_id: foxess_inv1_pv2_daily
    source: sensor.pv2_sum
    cycle: daily
  ct2_daily:
    unique_id: foxess_inv1_pv2_daily
    source: sensor.ct2_sum
    cycle: daily

**Reply by canton7** [reply]
Yes, that sort of thing would create 3 separate daily totals for PV1, PV2 and CT.

In `ct2_daily`, you probably want `unique_id: foxess_inv1_ct2_daily`

We're planning to provide built-in sensors for total PV1 and PV2 at some point, but until then it's best to create your own.

**Reply by LukeMc8** [reply]
Perfect thanks. I owe you guys a coffee.

**Reply by nathanmarlor** [reply]
BTW if you want a sensor to add PV1, PV2 and CT2 together it'd look like this (double check sensor names...!):

```
sensor:
  - platform: template
    sensors:
      sum_of_generation:
        friendly_name: "Sum of Generation"
        value_template: "{{ states('sensor.pv1_power')|int + states('sensor.pv2_power')|int + states('sensor.ct2')|int }}"
```

**Reply by LukeMc8** [reply]
Thanks, I cant decide if i should combine it all in one or split it. At the moment PV1 and PV2 are both South facing and CT2 is an East facing array so i have added two lots of solar production South facing (PV1 & 2 combined) and East facing. 

Thanks again for all the help.

**Reply by canton7** [reply]
If you want to combine PV1 and PV2, then the latest beta provides a Solar Generation Today sensor which is the sum of PV1 and PV2 today, as reported by the inverter.

**Reply by LukeMc8** [reply]
Yeah, thats the one I've used and then just created CT2 for now. It seems to be working.
